src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java

```diff
@@ -0,0 +1,392 @@
+        VCFHeader header = VCFHeaderReader.readHeaderFrom(new SeekablePathStream(vcfFile.toPath()));
+        stream.close();
+        return header;
+    }
+
+    private static <T> void assertForEachElementInLists(final List<T> actual, final List<T> expected, final BiConsumer<T, T> assertion) {
+        Assert.assertEquals(actual.size(), expected.size(), "different number of elements in lists:\n"
+                + actual.stream().map(Object::toString).collect(Collectors.joining("\n","actual:\n","\n"))
+                +  expected.stream().map(Object::toString).collect(Collectors.joining("\n","expected:\n","\n")));
+        for (int i = 0; i < actual.size(); i++) {
+
+            assertion.accept(actual.get(i), expected.get(i));
+        }
+    }
+
+    //==================================================================================================================
+    // Tests
+    //==================================================================================================================
+
+    @Test
+    public void GATK3LargeConcoranceTest() throws IOException {
+        assertVariantContextsMatch(getTestFile("HCOutput.NoAnnotations.vcf"), new File(getToolTestDataDir() + "expected/integrationTest.vcf"), Arrays.asList("-G", "Standard", "-G", "AS_Standard", "-L", "20:10000000-10100000", "-I", NA12878_20_21_WGS_bam), b37_reference_20_21);
+    }
+
+    @Test
+    public void testHasAnnotsNotAsking() throws IOException {
+        final File expected = new File(getToolTestDataDir() + "expected/testHsAnnotsNotAsking1.vcf");
+        final VCFHeader header = getHeaderFromFile(expected);
+        runVariantAnnotatorAndAssertSomething(getTestFile("vcfexamplemultisample.vcf"), new File(getToolTestDataDir() + "expected/testHsAnnotsNotAsking1.vcf"), Arrays.asList( "-I", largeFileTestDir + "CEUTrio.multisample.b37.1M-1M50k.bam"),
+                (a, e) -> {
+                    // We need to filter out sites where we saw a DP of 250 because we are comparing the results to GATK3, which downsamples to 250 reads per sample, which GATK4 does not currently support.
+                    if (!e.getGenotypes().stream().anyMatch(g -> g.hasDP() && g.getDP() >= 250)) {
+                        VariantContextTestUtils.assertVariantContextsAreEqualAlleleOrderIndependent(a, e, ATTRIBUTES_TO_IGNORE, header);
+                    }
+                },
+                b37_reference_20_21);
+    }
+
+    @Test
+    public void testHasAnnotsAsking() throws IOException {
+        final File expected = new File(getToolTestDataDir() + "expected/testHasAnnotsAsking1.vcf");
+        final VCFHeader header = getHeaderFromFile(expected);
+        runVariantAnnotatorAndAssertSomething(getTestFile("vcfexamplemultisample.vcf"), new File(getToolTestDataDir() + "expected/testHasAnnotsAsking1.vcf"), Arrays.asList("-G", "Standard", "-I", largeFileTestDir + "CEUTrio.multisample.b37.1M-1M50k.bam"),
+                (a, e) -> {
```

Comment:
factor out this method since it's repeated 3 times

Reply:
also, intellij suggested that instead of `!anyMatch` we might prefer `noneMatch`

Reply:
actually, this might be a good candidate for a dataprovider since there are a bunch of test that call the same code but use different data